src/push-notifications/use-push-notifications.tsx

```diff
@@ -0,0 +1,130 @@
+        .catch((e) => {
+          setIsError(true);
+          console.error(e);
+        })
+        .finally(() => setIsLoadingPermissionStatus(false));
+    } else if (Platform.OS === 'android') {
+      const platformVersion = Platform.Version; // Extracted outside promise to make typescript understand it's a number since Platform === 'android'
+      PermissionsAndroid.check(
+        PermissionsAndroid.PERMISSIONS.POST_NOTIFICATIONS,
+      )
+        .then((hasPermission) => {
+          // On SDK versions < 33 permission is not required, yet check() will return 'false' for those devices.
+          // However, getToken() will yield a device token for sdk < 33 even if check() returns 'false'.
+          setPermissionStatus(
+            hasPermission || platformVersion < 33 ? 'granted' : 'denied',
+          );
+        })
+        .catch((e) => {
+          setIsError(true);
+          console.error(e);
+        })
+        .finally(() => setIsLoadingPermissionStatus(false));
+    } else {
+      setIsLoadingPermissionStatus(false);
+      setIsError(true);
+    }
+  }, [Platform, messaging, PermissionsAndroid]);
+
+  const register = async () => {
+    const permissionStatus = await requestUserPermission();
+    setPermissionStatus(permissionStatus);
+    const token = await messaging().getToken();
+    if (!token) {
+      setIsError(true);
+      return;
+    }
+    registerMutation.mutate({token, language});
+  };
+
+  return {
+    isError:
+      isError ||
+      registerMutation.isError ||
+      configQuery.isError ||
+      configMutation.isError,
+    isLoading: isLoadingPermissionStatus || configQuery.isInitialLoading,
+    isUpdating: registerMutation.isLoading || configMutation.isLoading,
+    permissionStatus,
+    config: configQuery.data,
+    updateConfig: configMutation.mutate,
+    checkPermissions,
+    register,
+  };
+};
+
+async function requestUserPermission(): Promise<PermissionStatus> {
+  if (Platform.OS === 'ios') {
+    const authStatus = await messaging().requestPermission();
+    return mapIosPermissionStatus(authStatus);
+  } else if (Platform.OS === 'android') {
+    const permissionStatus = await PermissionsAndroid.request(
+      PermissionsAndroid.PERMISSIONS.POST_NOTIFICATIONS,
+    );
+    // Request permission for API level 33+. Not required for 32 or below.
+    return Platform.Version < 33
+      ? 'granted'
+      : mapAndroidPermissionStatus(permissionStatus);
+  }
+  return 'undetermined';
+}
+
+function mapIosPermissionStatus(
+  authStatus: FirebaseMessagingTypes.AuthorizationStatus,
+) {
+  switch (authStatus) {
+    case messaging.AuthorizationStatus.AUTHORIZED:
+    case messaging.AuthorizationStatus.PROVISIONAL:
+    case messaging.AuthorizationStatus.EPHEMERAL:
+      return 'granted';
+    case messaging.AuthorizationStatus.DENIED:
+      return 'denied';
+    case messaging.AuthorizationStatus.NOT_DETERMINED:
+      return 'undetermined';
+    default:
+      return 'undetermined';
```

Comment:
Should just skip the `default` clause, so the switch statement is exhaustive. Then typescript will warn us if new `AuthorizationStatus' enums are added in later versions of Firebase Messaging.